tests/test_validation_mismatch.py

```diff
@@ -0,0 +1,781 @@
+# SUBROUTINES RELATED TO WORKFLOW UNIT TESTING
+# from tests import config_tests
+import types
+from mock import Mock
+import sys
+from . import config_tests
+from sphinxval.utils import config
+
+module_name = 'config'
+mocked_config = types.ModuleType(module_name)
+sys.modules[module_name] = mocked_config
+mocked_config = Mock(name='sphinxval.utils.' + module_name)
+
+from sphinxval.utils import units_handler as vunits
+from sphinxval.utils import validation as validate
+from sphinxval.utils import metrics
+from sphinxval.utils import time_profile as profile
+from sphinxval.utils import resume
+from sphinxval.utils import plotting_tools as plt_tools
+from sphinxval.utils import match
+from sphinxval.utils import validation_json_handler as vjson
+from sphinxval.utils import classes as cl
+from sphinxval.utils import object_handler as objh
+
+
+
+
+from astropy import units
+import unittest
+import sys
+import pprint
+import pandas as pd
+import os
+import csv
+import logging
+import logging.config
+import pathlib
+import json
+from datetime import datetime
+
+from unittest.mock import patch
+import shutil # using this to delete the contents of the output folder each run - since the unittest is based on the existence/creation of certain files each loop
+
+
+
+logger = logging.getLogger(__name__)
+
+
+
+"""
+Mismatch unit test requires a slightly different 
+set up from the regular validation unit test
+since the dataframe takes different values
+from the config file (which I mock so that I can
+more easily control the values of). Also since
+this is mismatch the output file names may contain
+_mm at the end 
+"""
+
+
+def utility_setup_logging():
```

Comment:
Duplicated from test_validation.py -- move to common module and import e.g. `tests.utils`